packages/native-stack/src/views/NativeStackView.native.tsx

```diff
@@ -541,7 +542,12 @@ const SceneView = ({
                     ? !isRemovePrevented
                     : headerBackButtonMenuEnabled
                 }
-                headerShown={header !== undefined ? false : headerShown}
+                headerShown={
+                  header !== undefined ||
+                  (presentation === 'formSheet' && Platform.OS === 'android')
```

Comment:
had to add this line in order to fix `formSheet` layout having wrong height on android:
<img width="313" alt="Screenshot 2024-10-21 at 12 18 04" src="https://github.com/user-attachments/assets/9e455e42-7371-4f37-af6d-91df39230c65">